File: 66-plus-one/plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        c = 1
        for i in range(len(digits)-1, -1, -1):
            digits[i] += c
            c=0
            if digits[i] > 9:
                c = 1
                digits[i] = 0
        if c == 1:
            digits.insert(0, c)
        return digits

File: 66-plus-one/plus-one.go
func plusOne(digits []int) []int {
    c := 1
    for i := len(digits)-1; i > -1; i--{
        digits[i] +=1
        c = 0
        if digits[i] > 9{
            digits[i] = 0
            c = 1
        } else {
            return digits
        }
    }
    if c == 1{
        digits = append([]int{c}, digits...)
    }
    return digits
}